core/swapper.py

```diff
@@ -21,14 +21,29 @@ def process_video(source_img, frame_paths):
     with tqdm(total=len(frame_paths), desc="Processing", unit="frame", dynamic_ncols=True, bar_format='{l_bar}{bar}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_fmt}{postfix}]') as progress:
         for frame_path in frame_paths:
             frame = cv2.imread(frame_path)
+            swapper = get_face_swapper()
             try:
-                face = get_face(frame)
-                if face:
-                    result = get_face_swapper().get(frame, face, source_face, paste_back=True)
+                if core.globals.all_faces:
+                    all_faces = get_all_faces(frame)
+                    result = frame
+                    if len(all_faces) > 0:
+                        for singleFace in all_faces:
+                            if singleFace:
```

Comment:
I think if you rewrite the code a bit you can extract a method here that contains the P and S conditions

can be reused in the one face block aswell